megatron/core/dist_checkpointing/serialization.py

```diff
@@ -57,6 +59,22 @@
 logger = logging.getLogger(__name__)
 
 
+def _save(obj: object, path: AnyPath):
+    fileobj = io.BytesIO()
```

Comment:
I'm worried 2-stage writing might affect performance in the baseline scenario (no cloud).
Can we make it optional as well?